src/Cardano/Wallet/Api/Types.hs

```diff
@@ -159,6 +177,33 @@ data WalletPutPassphraseData = WalletPutPassphraseData
     , newPassphrase :: !(ApiT (Passphrase "encryption"))
     } deriving (Eq, Generic, Show)
 
+data PostTransactionData = PostTransactionData
+    { targets :: !(NonEmpty ApiCoinSelection)
+    , passphrase :: !(ApiT (Passphrase "encryption"))
+    } deriving (Eq, Generic, Show)
+
+data ApiTransaction = Transaction
+    { id :: !(ApiT (Hash "Tx"))
+    , amount :: !(Quantity "lovelace" Natural)
+    , insertedAt :: !ApiBlockData
+    , depth :: !(Quantity "block" Natural)
+    , direction :: !(ApiT Direction)
+    , inputs :: !(NonEmpty ApiCoinSelection)
+    , outputs :: !(NonEmpty ApiCoinSelection)
```

Comment:
inputs and outputs could be sepparate datatypes here such as `TxInputs` and `TxOutputs` but I decided to merge datatype into one named `ApiCoinSelection`. I am not sure is this a good idea or not. They do represent the same idea but I opt for the simplicity here and dropped the extra type safety that we were getting from having two types at this level

Alternitavely we would create two types `ApiTxInputs` and `ApiTxOutputs` which would look similar to `TxIn` and `TxOut` from Primitive module (except they will have wrappers for `ApiT`). Simiarly we would use `ApiTxOut` in `PostTransactionData.targets`.

Also `ApiCoinSelection` might be unfortunate name pick as it conflicts with coin selection algorithm that we implemented https://github.com/input-output-hk/cardano-wallet/tree/5c7d690e5d994c931e5b437c5bdbaf222f6dc675/src/Cardano/Wallet/CoinSelection although the context is the same

Reply:
Something other than `ApiCoinSelection` 👍 

If possible, a short doc-comment about what it is would be awesome too. E.g _if_ you were to go with `ApiResolvedInputs`, it might be worth knowing that it's used for `outputs`, and not just `inputs` of `ApiTransaction`

`ApiCoins` sounds reasonable to me. Or with some tweak like `ApiOwnedCoins` to indicate that they contain addresses, which I presume are the previous and current owner.